File: 2021/day1/p2.py
with open("input.txt", encoding="utf-8") as f:
	data = [int(i) for i in f.readlines()]
step = 3
count = 0
for i in range(len(data)-step):
	if data[i+step] > data[i]:
		count += 1
print(count)

File: 2021/day1/P2.java
import java.io.File;  // Import the File class
import java.util.ArrayList;
import java.util.Scanner;


public class P2 {
	public static void main(String[] args) throws Exception{
		ArrayList<Integer> data = new ArrayList<>();
		File file = new File("input.txt");
		Scanner reader = new Scanner(file);
		int score = 0;
		int step = 3;
		while (reader.hasNextLine()) {
			data.add(Integer.parseInt(reader.nextLine()));
		}
		for (int i=0; i<data.size()-step; ++i) {
			if (data.get(i) < data.get(i+step)) {
				++score;
			}
		}
		System.out.println(score);
	}
}